RIOT-OS/RIOT
# boards/esp32: changes the approach for peripheral configurations in ESP32 board definitions

### Contribution description

This PR changes the approach of peripheral configurations for ADC, DAC, I2C, SPI and UART in board definitions to the usual RIOT approach. With these changes, peripheral configurations use static const arrays in the `boards/esp32*/periph_conf.h` files and define the `*_NUMOF` macros using the size of these static array.

The static configuration arrays contain only definitions that can be changed by the board definition or the application. They do not contain any MCU implementation detail. The board definitions use preprocessor defines as before to fill these static configuration arrays. This makes it possible to override all configurations either with the make command or application specific configuration files.

### Testing procedure

Compilation and test with the most common ESP32 board should be executed
```
make BOARD=esp32-wroom-32 -C tests/periph_adc flash test
make BOARD=esp32-wroom-32 -C tests/periph_dac flash test
make BOARD=esp32-wroom-32 -C tests/periph_i2c flash test
make BOARD=esp32-wroom-32 -C tests/periph_pwm flash test
make BOARD=esp32-wroom-32 -C tests/periph_spi flash test
make BOARD=esp32-wroom-32 -C tests/periph_uart flash test
```
In addition to the tests above, the PR has been already tested with `tests/driver_lis3dh` and SPI as well as `tests/driver_lis3mdl` and I2C.

### Issues/PRs references

This PR contains also the changes already provided by PRs #10643 and #10641.
